ferminet/networks.py

```diff
@@ -440,17 +440,30 @@ def logdet_matmul(xs: Sequence[jnp.ndarray],
     determinant (or product of the i-th determinant in each spin channel, if
     full_det is not used).
   """
-  slogdets = [slogdet(x) for x in xs]
-  sign_in, logdet = functools.reduce(
-      lambda a, b: (a[0]*b[0], a[1]+b[1]), slogdets)
+  # Special case if there is only one electron in any channel
+  # We can avoid the log(0) issue by not going into the log domain
+  dets = [x.reshape(-1) for x in xs if x.shape[-1] == 1]
+  dets = functools.reduce(
+    lambda a, b: a*b, dets
+  ) if len(dets) > 0 else 1
+
+  slogdets = [slogdet(x) for x in xs if x.shape[-1] > 1]
```

Comment:
Can we also move this into the if branch? The test can then be `if any(x.shape[-1] for x in in xs):`

Reply:
Yes, will update